device/device.py

```diff
@@ -93,14 +97,37 @@ def get_supported_devices():
             Query for device support list
             :return: list of supported devices
         """
+        
+        # Remove at* prefix on all targets (e.g attiny202 -> tiny202)
+        # for legacy naming support
+        DEVICES_TINY_2K  = set([s[2:] for s in DEVICES_ATTINY_2K])
+        DEVICES_TINY_4K  = set([s[2:] for s in DEVICES_ATTINY_4K])
+        DEVICES_TINY_8K  = set([s[2:] for s in DEVICES_ATTINY_8K])
+        DEVICES_TINY_16K = set([s[2:] for s in DEVICES_ATTINY_16K])
+        DEVICES_TINY_32K = set([s[2:] for s in DEVICES_ATTINY_32K])
+        DEVICES_MEGA_8K  = set([s[2:] for s in DEVICES_ATMEGA_8K])
+        DEVICES_MEGA_16K = set([s[2:] for s in DEVICES_ATMEGA_16K])
+        DEVICES_MEGA_32K = set([s[2:] for s in DEVICES_ATMEGA_32K])
+        DEVICES_MEGA_48K = set([s[2:] for s in DEVICES_ATMEGA_48K])
+
+
         return sorted(
-            DEVICES_TINY_2K |
-            DEVICES_TINY_4K |
-            DEVICES_TINY_8K |
-            DEVICES_TINY_16K |
-            DEVICES_TINY_32K |
-            DEVICES_MEGA_8K |
-            DEVICES_MEGA_16K |
-            DEVICES_MEGA_32K |
-            DEVICES_MEGA_48K |
+            DEVICES_ATTINY_2K  |
+            DEVICES_TINY_2K    |
```

Comment:
Is the non-prepended device names necessary here, since you prepend the names further up?

Reply:
Yes, it is, because `get_supported_devices()` gets called first to check if the passed target matched any of the targets in the DEVICES* lists. If true, then continue on with `__init__(self, device_name)`.

If I comment out `DEVICES_TINY_2K`, it will no longer accept `tiny202` as target name.

This means that the following code can't be executed since it will never really get there:

https://github.com/mraardvark/pyupdi/blob/c3d2486ad6fcd2b9d9f03710ef3601f1629bbc88/device/device.py#L87-L88